fetch_news.py

```diff
@@ -124,9 +124,33 @@ def get_mean_embedding(mongo_client = None, connection_string = None):
 
     return m
 
-def delete_old_article_text(mongo_client):
+def trim_old_articles(mongo_client):
```

Comment:
This function is now doing multiple things. I would suggest splitting it into two functions: `delete_old_articles` and `trim_old_articles` which I think reflects the two major operations here better.